src/main/java/com/meilisearch/sdk/Client.java

```diff
@@ -242,4 +248,59 @@ public Key createKey(Key options) throws Exception {
     public void deleteKey(String key) throws Exception {
         this.keysHandler.deleteKey(key);
     }
+
+    public String generateTenantToken(Map<String, Object> searchRules) throws MeiliSearchException {
+        return this.generateTenantToken(searchRules, new TenantTokenOptions());
+    }
+
+    /**
+     * Generate a tenant token
+     *
+     * @param searchRules A Map of string, object which contains the rules to be enforced at search
+     *     time for all or specific accessible indexes for the signing API Key.
+     * @param options A TenantTokenOptions, the following fileds are accepted: - apiKey: String
+     *     containing the API key parent of the token. If you leave it empty the client API Key will
+     *     be used. - expiresAt: A DateTime when the key will expire. Note that if an expiresAt
+     *     value is included it should be in UTC time.
+     * @return String containing the tenant token
+     * @throws MeiliSearchException if an error occurs
+     */
+    public String generateTenantToken(Map<String, Object> searchRules, TenantTokenOptions options)
+            throws MeiliSearchException {
+        // Validate all fields
+        Date now = new Date();
+        TimeZone.setDefault(TimeZone.getTimeZone("UTC"));
+        if (options.getExpiresAt() != null && now.after((Date) options.getExpiresAt())) {
+            throw new MeiliSearchException("The date expiresAt should be in the future.");
+        }
+        if ((options.getApiKey() == null || options.getApiKey() == "")
+                && (this.config.apiKey == "" || this.config.apiKey == "")) {
+            throw new MeiliSearchException(
+                    "An api key is required in the client or should be passed as an argument.");
+        }
+        if (searchRules == null) {
+            throw new MeiliSearchException(
+                    "The searchRules field is mandatory and should be defined.");
+        }
+
+        String secret;
+        if (options.getApiKey() == null || options.getApiKey() == "") {
```

Comment:
This validation repeats 4 times in this code, I'm not sure if it happens outside of this scope as well. But we could create a small function to help clean that out!

```java
private boolean isNullOrEmpty(String str) {
   return str == null || str.trim().length() == 0;
}
```

Or use `com.google.common.base.Strings.isNullOrEmpty(String string)` (but I'm not sure how guava is used in the meilisearch-java, since I just found it in the deps but not in the code).

Reply:
I just find 2 of it and I modify the code to check the length